src/components/dialogs/network-modifications/voltage-level-topology-modification/voltage-level-topology-modification-form.tsx

```diff
@@ -0,0 +1,221 @@
+                },
+            },
+            {
+                field: PREV_CONNECTION_STATUS,
+                flex: 1,
+                headerComponent: HeaderWithTooltip,
+                cellRenderer: (params: { data?: any; node?: any }) => {
+                    if (params.data.type === 'SEPARATOR') {
+                        return null;
+                    } else {
+                        return intl.formatMessage({ id: params.data[PREV_CONNECTION_STATUS] });
+                    }
+                },
+                headerComponentParams: {
+                    displayName: intl.formatMessage({ id: 'previousStatus' }),
+                    tooltipTitle: intl.formatMessage({
+                        id: isNodeBuilt(currentNode) ? 'builtNodeTooltipVlTopoModif' : 'notBuiltNodeTooltipVlTopoModif',
+                    }),
+                    isNodeBuilt: isNodeBuilt(currentNode),
+                    disabledTooltip: !isUpdate && isNodeBuilt(currentNode),
+                },
+            },
+            {
+                field: CURRENT_CONNECTION_STATUS,
+                flex: 1,
+                cellRenderer: (params: { data?: any; node?: any }) => {
+                    if (params.data.type === 'SEPARATOR') {
+                        return null;
+                    }
+                    const watchTable: SwitchRowForm[] = getValues(TOPOLOGY_MODIFICATION_TABLE);
+                    const formIndex = watchTable.findIndex(
+                        (item: SwitchRowForm) => item.switchId === params.data.switchId
+                    );
+                    return BooleanNullableCellRenderer({
+                        name: `${TOPOLOGY_MODIFICATION_TABLE}[${formIndex}].${CURRENT_CONNECTION_STATUS}`,
+                    });
+                },
+                headerComponent: HeaderWithTooltip,
+                headerComponentParams: {
+                    displayName: intl.formatMessage({ id: 'currentStatus' }),
+                    tooltipTitle: intl.formatMessage({
+                        id: isNodeBuilt(currentNode) ? 'builtNodeTooltipVlTopoModif' : 'notBuiltNodeTooltipVlTopoModif',
+                    }),
+                    isNodeBuilt: isNodeBuilt(currentNode),
+                    disabledTooltip: true,
+                },
+                editable: false,
+            },
+        ],
+        [currentNode, intl, isUpdate, getValues]
+    );
+
+    const copyPreviousToCurrentStatus = useCallback(() => {
+        const formValues = getValues(TOPOLOGY_MODIFICATION_TABLE);
+        formValues.forEach((row: SwitchRowForm, index: number) => {
+            if (row.type === 'SEPARATOR') {
+                return;
+            }
+            const newValue =
+                row[PREV_CONNECTION_STATUS] === 'Open' ? false : row[PREV_CONNECTION_STATUS] === 'Closed' ? true : null;
+            setValue(`${TOPOLOGY_MODIFICATION_TABLE}[${index}].${CURRENT_CONNECTION_STATUS}`, newValue);
+        });
+    }, [getValues, setValue]);
```

Comment:
to move on dialog to make the form component more pure

Reply:
setValue of useForm does not notify event with therre options: shouldValidate, shouldDirty, shouldTouch
It seems this method setValue is suitable for initialize form

=> to keep copyPreviousToCurrentStatus in form to use setValue of useFormContext which notify to component